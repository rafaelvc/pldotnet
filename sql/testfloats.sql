-- Float4 (real): 6 digits of precison
CREATE OR REPLACE FUNCTION returnReal() RETURNS real AS $$
return 1.50055f;
$$ LANGUAGE pldotnet;
SELECT returnReal();

CREATE OR REPLACE FUNCTION sumReal(a real, b real) RETURNS real AS $$
return a+b;
$$ LANGUAGE pldotnet;
SELECT sumReal(1.50055, 1.50054); -- 3.00109

--- Float8 (double precision): 15 digits of precison
CREATE OR REPLACE FUNCTION returnDouble() RETURNS double precision AS $$
return 11.0050000000005;
$$ LANGUAGE pldotnet;
SELECT returnDouble();

CREATE OR REPLACE FUNCTION sumDouble(a double precision, b double precision) RETURNS double precision AS $$
return a+b;
$$ LANGUAGE pldotnet;
SELECT sumDouble(10.5000000000055, 10.5000000000054); -- 21.0000000000109
