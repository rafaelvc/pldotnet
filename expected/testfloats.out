-- Float4 (real): 6 digits of precison
CREATE OR REPLACE FUNCTION returnReal() RETURNS real AS $$
return 1.50055f;
$$ LANGUAGE pldotnet;
SELECT returnReal();
 returnreal 
------------
    1.50055
(1 row)

CREATE OR REPLACE FUNCTION sumReal(a real, b real) RETURNS real AS $$
return a+b;
$$ LANGUAGE pldotnet;
SELECT sumReal(1.50055, 1.50054); -- 3.00109
 sumreal 
---------
 3.00109
(1 row)

--- Float8 (double precision): 15 digits of precison
CREATE OR REPLACE FUNCTION returnDouble() RETURNS double precision AS $$
return 11.0050000000005;
$$ LANGUAGE pldotnet;
SELECT returnDouble();
   returndouble   
------------------
 11.0050000000005
(1 row)

CREATE OR REPLACE FUNCTION sumDouble(a double precision, b double precision) RETURNS double precision AS $$
return a+b;
$$ LANGUAGE pldotnet;
SELECT sumDouble(10.5000000000055, 10.5000000000054); -- 21.0000000000109
    sumdouble     
------------------
 21.0000000000109
(1 row)

